includes/class-newspack-segments-migration.php

```diff
@@ -262,7 +282,8 @@ public static function migrate_user_data() {
 		if (
 			! is_user_logged_in() ||
 			get_user_meta( get_current_user_id(), 'newspack_popups_reader_data_migrated', true ) ||
-			! class_exists( 'Newspack\Reader_Data' )
+			! class_exists( 'Newspack\Reader_Data' ) ||
+			! self::$legacy_table_exists
```

Comment:
```suggestion
			! self::check_for_legacy_table()
```

To avoid running the `check_for_legacy_table()` method (and a db query) on every page load, the method could cache `self::$legacy_table_exists` and return if the value has been set.

Reply:
5f2f118 caches the result in a new options row—did you have another method in mind?

Reply:
My original idea was just caching in memory, but that also works 🙂

Reply:
Why run the method on `init` instead of calling the method when needed inside the migration method?

Reply:
> Why run the method on `init` instead of calling the method when needed inside the migration method?

That was an oversight, sorry! 181aff0 removes callback on `init` and only runs the method when needed (when a reader registers or logs in).

> My original idea was just caching in memory, but that also works 🙂

I tried this, but the cached var didn't persist across new pageview requests, so it ended up running the DB query on every request anyway. The other thing I was going to try was a transient, but an option makes the most sense to me because it's unlikely that the existence of the legacy tables will change (until we drop them).